Ожидает проверки

WorldOfBatle opened the pull request.

WorldOfBattle committed:
Подготовка к работе

WorldOfBattle committed:
Созданы заготовки функций f() и t()

WorldOfBattle committed:
Реализована функция f() для длины отрезка

WorldOfBattle committed:
Реализована функция t() для проверки существования треугольника

WorldOfBattle committed:
Добавлен код проверки построения треугольника по 3 точкам

tak333 commented on HW-2/main.cpp:
```diff
@@ -0,0 +1,42 @@
+    std::cout << "Длина стороны b = " << b << std::endl;
+    std::cout << "Длина стороны c = " << c << std::endl;
+
+    std::cout << (t(a, b, c) ? "Треугольник с такими сторонами существует." : "Треугольник с такими сторонами не существует.");
```
Одинаковые части фразы лучше вынести из условного оператора для краткости.

WorldOfBatle replied:
Вывел

tak333 commented on HW-2/main.cpp:
```diff
@@ -0,0 +1,42 @@
+    std::cout << "Длина стороны b = " << b << std::endl;
+    std::cout << "Длина стороны c = " << c << std::endl;
+
+    std::cout << (t(a, b, c) ? "Треугольник с такими сторонами существует." : "Треугольник с такими сторонами не существует.");
```
Зачем повторный вызов функции t?

WorldOfBatle replied:
На автомате перекопипастил из строчки и из-за этого потом и условие кривое сделал
Исправлено

tak333 commented on HW-2/main.cpp:
```diff
@@ -0,0 +1,42 @@
+#include <cmath> // Для sqrt, pow
+
+// Функция f(x1, y1, x2, y2)
+double f(double x1, double y1, double x2, double y2)
```
Прошу прощения за неточность, давайте в этом задании соблюдать правила именования функций согласно принципам "хорошего" кода.
Комментарий в д/з внесла.

WorldOfBatle replied:
Просто в самом задании были указаны для функций такие имена, поэтому их и использовал
Исправлено

tak333 commented on HW-2/main.cpp:
```diff
@@ -0,0 +1,42 @@
+}
+
+// Функция t(a, b, c)
+bool t(double a, double b, double c)
```
Аналогично, давайте сделаем имя функции говорящим.

WorldOfBatle replied:
Исправлено

tak333 commented on HW-2/main.cpp:
```diff
@@ -0,0 +1,42 @@
+// Функция t(a, b, c)
+bool t(double a, double b, double c)
+{
+    return ((a + b > c) && (b + c > a) && (a + c > b)) ? true : false;
```
В данном случае вы проверяете, является ли выражение true и в таком случае возвращаете true, а если значение выражение равно false, то вы выбираете возвращать false.
Можно ли как-то схитрить и упростить это?

WorldOfBatle replied:
Использовал переменную, всё на свои места встало)
Поправил условие

tak333 replied:
Вот тут как раз не поправили: 

> В данном случае вы проверяете, является ли выражение true и в таком случае возвращаете true, а если значение выражение равно false, то вы выбираете возвращать false.
> Можно ли как-то схитрить и упростить это?

Данный комментарий относится к данной строке кода:
`return ((a + b > c) && (b + c > a) && (a + c > b)) ? true : false;`

Понимаете, о чём я ?

P.S. после этого исправления кидайте снова запрос на слияние.

WorldOfBatle replied:
Убрал лишний условный оператор в функции canFormTriangle, т.к. там и так было приведение типа в бул
Верно?

tak333 reviewed (changes requested):
Очень хорошо, красиво и читаемо, осталась пара моментов!

WorldOfBattle committed:
Внесены изменения:
1. Убран повторяющийся текст из условного оператора.
2. Убран повторный вызов функции проверки сущестования треуголника, используется ранее созданная переменная.
3. Фунцкции переименованны, в соответствии с правилами хорошего кода (ранее так назвал, так как они в условии задачи так именовались)
4. Так как теперь используется переменная, сделал пряморукое условие

tak333 reviewed (changes requested):
Добейте одну строчку и после этого кидайте снова запрос на слияние.

WorldOfBattle committed:
Убрал лишний условный оператор в функции canFormTriangle, т.к. там и так было приведение типа в бул

WorldOfBatle commented:
Готово
